Sync/SyncTelemetry.swift

```diff
@@ -0,0 +1,113 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+import Account
+
+public enum SyncReason: String {
+    case Startup = "startup"
+    case Scheduled = "scheduled"
+    case Backgrounded = "backgrounded"
+    case User = "user"
```

Comment:
Call this `SyncNow`? There's a separate code path for pull-to-refresh in the History panel.